lte/gateway/c/oai/tasks/s1ap/s1ap_mme_handlers.c

```diff
@@ -1922,125 +1917,122 @@ static bool construct_s1ap_mme_full_reset_req(
-  s1ap_timer_arg_t timer_arg = {0};
+    s1ap_state_t* state, const sctp_assoc_id_t assoc_id, bool reset) {
+  arg_s1ap_send_enb_dereg_ind_t arg  = {0};
+  uint8_t itrn                       = 0;
+  MessageDef* message_p              = NULL;
+  enb_description_t* enb_association = NULL;
+  s1ap_timer_arg_t timer_arg         = {0};
 
   OAILOG_FUNC_IN(LOG_S1AP);
-  /*
-   * Checking that the assoc id has a valid eNB attached to.
-   */
-  enb_association = s1ap_state_get_enb(state, assoc_id);
+
   OAILOG_INFO(
-    LOG_S1AP,
-    "SCTP disconnection request for association id %u. Reset Flag = %u \n",
-    assoc_id,
-    reset);
+      LOG_S1AP,
+      "SCTP disconnection request for association id %u, Reset Flag = %u \n",
+      assoc_id, reset);
 
+  // Checking if the assoc id has a valid eNB attached to it
+  enb_association = s1ap_state_get_enb(state, assoc_id);
   if (enb_association == NULL) {
     OAILOG_ERROR(LOG_S1AP, "No eNB attached to this assoc_id: %d\n", assoc_id);
     OAILOG_FUNC_RETURN(LOG_S1AP, RETURNerror);
   }
 
-  OAILOG_INFO(
-    LOG_S1AP,
-    "SCTP disconnection request for association id %u. Reset Flag = "
-    "%u.Connected UEs = %d \n",
-    assoc_id,
-    reset,
-    enb_association->nb_ue_associated);
-  // First check if we can just reset the eNB state if there are no UEs.
+  // First check if we can just reset the eNB state if there are no UEs
   if (!enb_association->nb_ue_associated) {
     if (reset) {
-      enb_association->s1_state = S1AP_INIT;
       OAILOG_INFO(
-        LOG_S1AP,
-        "SCTP reset request for association id %u. No Connected UEs.  = %u \n",
-        assoc_id,
-        reset);
+          LOG_S1AP,
+          "SCTP reset request for association id %u. No Connected UEs. "
+          "Reset Flag = %u\n",
+          assoc_id, reset);
+
       OAILOG_INFO(
-        LOG_S1AP,
-        "Moving eNB with association id %u to INIT state\n",
-        assoc_id);
+          LOG_S1AP, "Moving eNB with assoc_id %u to INIT state\n", assoc_id);
+      enb_association->s1_state = S1AP_INIT;
       update_mme_app_stats_connected_enb_sub();
     } else {
-      s1ap_remove_enb(state, enb_association);
-      update_mme_app_stats_connected_enb_sub();
       OAILOG_INFO(
-        LOG_S1AP,
-        "SCTP Shutdown request for association id %u. No Connected UEs.  = %u "
-        "\n",
-        assoc_id,
-        reset);
+          LOG_S1AP,
+          "SCTP Shutdown request for association id %u. No Connected UEs. "
+          "Reset Flag = %u\n",
+          assoc_id, reset);
+
       OAILOG_INFO(LOG_S1AP, "Removing eNB with association id %u \n", assoc_id);
+      s1ap_remove_enb(state, enb_association);
+      update_mme_app_stats_connected_enb_sub();
     }
     OAILOG_FUNC_RETURN(LOG_S1AP, RETURNok);
   }
 
+  OAILOG_INFO(
+      LOG_S1AP,
+      "SCTP disconnection request for association id %u, Reset Flag = "
+      "%u. Connected UEs = %d \n",
+      assoc_id, reset, enb_association->nb_ue_associated);
```

Comment:
merge this with the log at the top of this function. They are not very different

Reply:
Updated the PR